G1ANT.Addon.Net/Models/SimpleAuthenticationModel.cs

```
@@ -0,0 +1,39 @@
+﻿using MailKit.Net.Imap;
+using MailKit.Net.Smtp;
+using System.Net;
+
+namespace G1ANT.Addon.Net.Models
+{
+    public class SimpleAuthenticationModel : IAuthenticationModel
+    {
+        public string Username { get; set; } = "";
+        public string Password { get; set; } = "";
+
+        public SimpleAuthenticationModel()
+        { }
+
+        public SimpleAuthenticationModel(string username, string password)
+        {
+            Username = username;
+            Password = password;
+        }
+
+        public void Authenticate(ImapClient client)
+        {
+            if (!string.IsNullOrEmpty(Username) && !string.IsNullOrEmpty(Password))
+            {
+                var credentials = new NetworkCredential(Username, Password);
+                client.Authenticate(credentials);
+            }
+        }
+
+        public void Authenticate(SmtpClient client)
+        {
+            if (!string.IsNullOrEmpty(Username) && !string.IsNullOrEmpty(Password))
+            {
+                var credentials = new NetworkCredential(Username, Password);
+                client.Authenticate(credentials);
+            }
+        }
```

Is it ok if username or password is empty? or we should throw an exception?

---

it is ok in some cases when server doesn't need user/password (Rhenus case)